sunpy/coordinates/wcs_utils.py

```diff
@@ -60,6 +79,31 @@ def solar_wcs_frame_mapping(wcs):
         return Heliocentric(obstime=dateobs, observer=observer)
 
 
+def _set_wcs_aux_obs_coord(wcs, obs_frame):
+    """
+    Set (in-place) observer coordinate information on a WCS.
+
+    Parameters
+    ----------
+    wcs : astropy.wcs.WCS
+    obs_frame : astropy.coordinates.SkyCoord, astropy.coordinates.CoordinateFrame
+    """
+    # Sometimes obs_coord can be a frame (with data); convert to a SkyCoord to
+    # ensure the .frame attribute is present
+    if not isinstance(obs_frame, BaseCoordinateFrame):
+        obs_frame = obs_frame.frame
+
+    if isinstance(obs_frame, HeliographicStonyhurst):
+        wcs.wcs.aux.hgln_obs = obs_frame.lon.to_value(u.deg)
+    elif isinstance(obs_frame, HeliographicCarrington):
+        wcs.wcs.aux.crln_obs = obs_frame.lon.to_value(u.deg)
+    else:
+        raise ValueError('obs_coord must be in a Stonyhurst or Carrington frame')
```

Comment:
What's the reasoning for not converting here but doing so in mapbase?

Reply:
This provides a more stringent check on obs_frame, which doesn't let anything through that isn't stonyhurst/carrington, which is then the behaviour of `solar_frame_to_wcs_mapping`, which I think is desired (to error instead of transforming).

Then, if elsewhere (e.g. `Map`) we want to be more lenient, we can always transform to `stonyhurst` before setting this metadata to let any coordinate system through.

That's my thinking at least, open to other options.

Reply:
I suppose that makes some sense for this helper, if we consider this a straight up wrapper of the WCS functionality. On the other hand, the point of our coordinate infra is that the transforms shouldn't ever really need to be special cased, they will either work or they wont. So I am not sure what the reasoning would be to not want the transform done if you are calling this helper function :thinking:

Reply:
Maybe @ayshih can have a casting vote - I'm unsure what the best option is.

Reply:
Since this is a private function, and for the sake of getting this shipped, I would suggest leaving this as is and we can always revisit?

Reply:
Sounds good.